py/foundation/geometry.py

```diff
@@ -237,6 +237,13 @@ def intersection(bs: Iterable['BBox']) -> Optional['BBox']:
       return None
     return BBox(ix, iy)
 
+  @staticmethod
+  def union(bs: Iterable['BBox']) -> 'BBox':
+    b = BBox.spanning(
+      chain.from_iterable([b.corners() for b in bs]))
+    assert b is not None
```

Comment:
Should we instead do `if b is None: raise Exception`? In some Python environments, you can turn off assert statements to optimize runtime, but maybe that would be an intended behavior...

Reply:
Actually we should probably just propagate the `None`, and document that it happens if `bs` yields nothing aka passed an empty iterator.